src/io/xpi.js

```diff
@@ -103,62 +101,55 @@ export class Xpi extends IOBase {
     }
   }
 
-  getFileAsStream(path) {
+  async getFileAsStream(path) {
+    this.checkPath(path);
+    const zipfile = await this.open();
     return new Promise((resolve, reject) => {
-      this.checkPath(path);
-      return this.open()
-        .then((zipfile) => {
-          zipfile.openReadStream(this.files[path], (err, readStream) => {
-            if (err) {
-              return reject(err);
-            }
-            return resolve(readStream.pipe(stripBomStream()));
-          });
-        })
-        .catch(reject);
+      zipfile.openReadStream(this.files[path], (err, readStream) => {
+        if (err) {
+          return reject(err);
+        }
+        return resolve(readStream.pipe(stripBomStream()));
+      });
     });
   }
 
-  getFileAsString(path) {
-    return this.getFileAsStream(path)
-      .then((fileStream) => {
-        return new Promise((resolve, reject) => {
-          let buf = Buffer.from('');
-          fileStream.on('data', (chunk) => {
-            buf = Buffer.concat([buf, chunk]);
-          });
-
-          // Once the file is assembled, resolve the promise.
-          fileStream.on('end', () => {
-            const fileString = buf.toString('utf8');
-            resolve(fileString);
-          });
-
-          fileStream.on('error', reject);
-        });
+  async getFileAsString(path) {
+    const fileStream = await this.getFileAsStream(path);
+
+    return new Promise((resolve, reject) => {
+      let buf = Buffer.from('');
+      fileStream.on('data', (chunk) => {
+        buf = Buffer.concat([buf, chunk]);
+      });
+
+      // Once the file is assembled, resolve the promise.
+      fileStream.on('end', () => {
+        const fileString = buf.toString('utf8');
+        resolve(fileString);
       });
+
+      fileStream.on('error', reject);
+    });
   }
 
-  getChunkAsBuffer(path, chunkLength) {
+  async getChunkAsBuffer(path, chunkLength) {
+    this.checkPath(path);
+    const zipfile = await this.open();
     return new Promise((resolve, reject) => {
-      this.checkPath(path);
-      return this.open()
-        .then((zipfile) => {
-          // eslint-disable-next-line consistent-return
-          zipfile.openReadStream(this.files[path], (err, readStream) => {
-            if (err) {
-              return reject(err);
+      // eslint-disable-next-line consistent-return
```

Comment:
We should be able to remove this `eslint-disable-next-line` here by changing line 143 into:

```js
if (err) {
  reject(err);
  return;
}
```

Reply:
Done